common/setups/hybrid/gmm_system.py

```diff
@@ -264,7 +287,7 @@ def cart_and_lda(
     def triphone_training(
         self,
         name: str,
-        corpus: str,
+        corpus_key: str,
         feature_flow: str,
         initial_alignment: str,
```

Comment:
Maybe we use the `_key` suffix for all string variables that look into the member dicts?

Reply:
fine with me.

Reply:
I think this makes it clearer.

Reply:
I just looked into the code again and no this is not perfect.. as soon as we use the `meta.select_element` we can use both.. 

`feature_flow` being the selector.. so actually the type is not correct..

```
def select_element(collection, default_corpus, selector, default_index=-1):
    if not isinstance(selector, (list, tuple, str)):
        return selector
```

Reply:
I agree.

But sometimes a function accepts an `Object` or a (`str`,`str`,`int`) tupel of (corpus_key, name, index) and gets the correct `Object` from the member dicts via `i6_core.meta.system.select_element`.
How would we handle/name this?

Reply:
so depending on how we process this selector down the line it can be `str` or `str|rasr.SOMETHING`

Reply:
so I ll check the code and adapt where possible but in some cases adding the key would limit our choices artificially..

Reply:
maybe as an example to see what I mean, check `monophone_training` function, how `feature_energy_flow` and `feature_flow` are handled. Former has to be a `str`, latter does not.

Reply:
> But sometimes a function accepts an Object or a (str,str,int) tupel of (corpus_key, name, index) and gets the correct Object from the member dicts via i6_core.meta.system.select_element.
How would we handle/name this?

not sure if this is the best approach. I would leave it as `feature_flow` but add a type as: `Union[str, tuple[str, str,str], rasr.FlagDependentFlowAttribute`.
But maybe that gets too unwieldy?